x/leverage/keeper/internal_test.go

```diff
@@ -69,3 +69,42 @@ func (tk *TestKeeper) SetInterestScalar(ctx sdk.Context, denom string, scalar sd
 func (tk *TestKeeper) SetReserveAmount(ctx sdk.Context, coin sdk.Coin) error {
 	return tk.Keeper.setReserveAmount(ctx, coin)
 }
+
+// AllInvariants runs all invariants of the x/leverage module, including the inefficient ones.
+// This is only used in testing.
+func (tk *TestKeeper) AllInvariants() sdk.Invariant {
+	k := *tk.Keeper
+	return func(ctx sdk.Context) (string, bool) {
+		res, stop := ReserveAmountInvariant(k)(ctx)
+		if stop {
+			return res, stop
+		}
+
+		res, stop = CollateralAmountInvariant(k)(ctx)
+		if stop {
+			return res, stop
+		}
+
+		res, stop = BorrowAmountInvariant(k)(ctx)
+		if stop {
+			return res, stop
+		}
```

Comment:
maybe we can change `CollateralAmountInvariant` and `BorrowAmountInvariant` to be a function which directly execute the checks, rather than returns a function? `CollateralAmountInvariant(k, ctx)`

Reply:
For now I kept them in their `sdk.Invariant` form in case we want to feed them into any future tests as invariants.

Changed their names to `InefficientCollateralAmountInvariant` and `InefficientBorrowAmountInvariant ` and added notes to their godoc that they shouldn't be added in production.